config/service.go

```diff
@@ -276,6 +276,41 @@ func (c *ServiceConfigs) Validate() error {
 	return nil
 }
 
+// CTSUserDefinedMeta generates a map of service name to user defined metadata
+// from a list of service IDs or service names.
+func (c *ServiceConfigs) CTSUserDefinedMeta(serviceList []string) map[string]map[string]string {
+	if c == nil {
+		return nil
+	}
+
+	services := make(map[string]bool)
+	for _, s := range serviceList {
+		services[s] = true
+	}
+
+	m := make(map[string]map[string]string)
+	for _, s := range *c {
+		if len(s.CTSUserDefinedMeta) == 0 {
+			continue
+		}
+
+		serviceName := *s.Name
+		if s.ID != nil && *s.ID != "" {
```

Comment:
I came across `StringPresent()` previously, which I think you added! I think it could be reused here and potentially line 306 below to help readability.

Reply:
I love it, thanks for reminding me about `StringPresent`